csi/src/node/nvmfutil.rs

```diff
@@ -0,0 +1,179 @@
+use std::{io, thread, time};
+use udev::Enumerator;
+
+fn find_nvmf_device_by_uuid(str_uuid: &str) -> Result<String, String> {
+    let prop = "ID_WWN";
+    let value = format!("uuid.{}", str_uuid);
+
+    trace!("find_nvmf_device_by_uuid uuid={}", str_uuid);
+    let mut enumerator = Enumerator::new().unwrap();
+    enumerator.match_subsystem("block").unwrap();
+    //    enumerator.match_property("DEVTYPE", "disk").unwrap();
+    enumerator
+        .match_property("ID_MODEL", "MayaStor NVMF controller")
+        .unwrap();
+    for dev in enumerator.scan_devices().unwrap() {
+        if let Some(udev_value) = dev.property_value(prop) {
+            if udev_value.to_str().unwrap().contains(&value) {
+                trace!(
+                    "find_nvmf_device_by_uuid {} got {:?}",
+                    str_uuid,
+                    dev.property_value("DEVNAME")
+                );
+
+                return Ok(dev
+                    .property_value("DEVNAME")
+                    .unwrap()
+                    .to_str()
+                    .unwrap()
+                    .to_string());
+            }
+        }
+    }
+
+    // fall through
+    Err(format!("device not found for {}", str_uuid))
+}
+
+fn uuid_from_str(s: &str) -> String {
+    //TODO something more sane.
+    //hack { convert something like
+    //0)    nvmf://192.168.122.98:8420/nqn.2019-
+    //1)        05.io.openebs:nexus-
+    //2)        04089481-
+    //3)        cc69-
+    //4)        45c1-
+    //5)        8009-
+    //6)        8aa7167a633a
+    // to uuid string value.
+    // will return a garbage string if passed in garbage,
+    // but that is okay for now since this is only used
+    // to find devices with a matching uuid and nothing
+    // will match the garbage.
+    let mut frags = s.split('-').collect::<Vec<_>>();
+    while frags.len() > 5 {
+        frags.remove(0);
+    }
+    itertools::join(&frags, "-")
+    // } hack
+}
+
+fn wait_for_path_to_exist(uuid: String, max_retries: i32) -> Option<String> {
+    let second = time::Duration::from_millis(1000);
```

Comment:
Would it be better to use _time::Duration::from_secs(1)_?

Reply:
https://github.com/openebs/Mayastor/pull/302/commits/103a309bb8f0d25e4b4cf62544a0d24d400cd998